src/main/scala/strawman/collection/mutable/ArrayBuffer.scala

```diff
@@ -39,7 +39,7 @@ class ArrayBuffer[A] private (initElems: Array[AnyRef], initLength: Int)
         start = 0
       }
       else {
-        val newelems = new Array[AnyRef](end * 2)
+        val newelems = new Array[AnyRef](if (end == 0) 16 else (end * 2))
```

Comment:
The integer overflow isn't a terrible problem because you get an exception when you try to allocate -(1 << 31) bytes.  It's not the ideal exception to have, but you were going to get an exception anyway, and practically nobody checks whether something is a `NegativeArraySizeException` or whatever we'd throw on a too-full array.  But in general I fully agree that we need to keep careful watch on out of bounds errors and other such corner cases.